core/src/main/java/me/mykindos/betterpvp/core/combat/events/CustomDamageReductionEvent.java

```diff
@@ -1,15 +1,20 @@
 package me.mykindos.betterpvp.core.combat.events;
 
-import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.EqualsAndHashCode;
 import me.mykindos.betterpvp.core.framework.events.CustomEvent;
 
 @EqualsAndHashCode(callSuper = true)
 @Data
-@AllArgsConstructor
 public class CustomDamageReductionEvent extends CustomEvent {
 
     private final CustomDamageEvent customDamageEvent;
+    private final double initialDamage;
     private double damage;
+
+    public CustomDamageReductionEvent(CustomDamageEvent customDamageEvent, double damage) {
+        this.customDamageEvent = customDamageEvent;
+        this.damage = damage;
+        this.initialDamage = damage;
```

Comment:
Can't you just get this from the customdamageevent

Reply:
Actually, no. The main event's damage may be calculated after by a custom damage adapter like mythic mobs or any we introduce later. The `initialDamage` in this method saves the latest real damage the player takes before any modifier like armor reduction or other adapters is introduced.

Reply:
But isn't that the point of rawdamage in customdamageevent, to indicate what the initial damage was before any modifiers

Reply:
That's the raw damage of the attack. After that, comes modifiers like skills, then finally comes adapters and armor reduction. The modifiers is the `initialDamage` or the `finalDamage` of the custom damage events that is later overwritten by the adapters and armor reduction since there's no selective field for it.

Reply:
![Capture (12ORm)](https://github.com/Mykindos/BetterPvP/assets/40250304/f4a655b0-a24d-4b25-bf99-20c84a09de51)